--- slmodem-2.9.10/modem/modem.c~	2004-08-08 15:29:19.000000000 +0200
+++ slmodem-2.9.10/modem/modem.c	2005-07-02 21:25:37.564328848 +0200
@@ -1698,7 +1698,7 @@
 
 int modem_reset(struct modem *m)
 {
-	static int sregs_init(unsigned char sregs[]);
+	int sregs_init(unsigned char sregs[]);
 	MODEM_DBG("modem reset...\n");
 	if(m->state != STATE_MODEM_IDLE)
 		modem_hup(m,1);
@@ -1723,7 +1723,7 @@
  */
 
 /* set default init values */
-static int sregs_init(unsigned char sregs[])
+int sregs_init(unsigned char sregs[])
 {
         sregs[SREG_ESCAPE_CHAR]                  = '+' ; /* escape char */
         sregs[SREG_CR_CHAR]                      = '\r'; /* cr char */
